account_invoice_margin/report/account_invoice_report.py

```diff
@@ -1,14 +1,45 @@
 # © 2017 Sergio Teruel <sergio.teruel@tecnativa.com>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
-from odoo import fields, models
+from odoo import api, fields, models
 
 
 class AccountInvoiceReport(models.Model):
     _inherit = "account.invoice.report"
 
     margin = fields.Float(string="Margin", readonly=True)
+    margin_percent = fields.Float(string="Margin (%)", readonly=True)
 
     def _select(self):
         select_str = super()._select()
-        return "%s, SUM(line.margin_signed) AS margin" % select_str
+        return (
+            "%s, SUM(line.margin_signed) AS margin, "
+            "AVG(move.margin_percent)  AS margin_percent" % select_str
+        )
+
+    @api.model
+    def read_group(
+        self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True
+    ):
+        res = super(AccountInvoiceReport, self).read_group(
+            domain,
+            fields,
+            groupby,
+            offset=offset,
+            limit=limit,
+            orderby=orderby,
+            lazy=lazy,
+        )
+        if "margin" not in fields:
+            return res
+
+        full_fields = all(x in fields for x in {"margin", "margin_percent"})
+        margin_percent = "margin_percent" in fields
+        for line in res:
+            if full_fields:  # compute difference
+                line["margin_percent"] = (
+                    (line["margin"] or 0.0) / line["price_total"]
```

Comment:
Error when `line["price_total"]` = 0

Reply:
Yes. It also problem on 12. I created invoice with 0 amount.

Reply:
Put then `if full_fields and line["price_total"]:` as condition

Reply:
I like your solution compare to mine
> " ( line["price_total"] or 1.0) "